bindings/native/src/lib.rs

```diff
@@ -0,0 +1,340 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+extern crate core;
+
+mod shared;
+mod wrapper;
+
+//#![allow(unused_imports)]
+use log::LevelFilter;
+
+use std::{
+    cell::RefCell,
+    error::Error,
+    ffi::{CStr, CString},
+    os::raw::c_char,
+    ptr, slice,
+};
+
+use crate::{
+    shared::hash_blake2b,
+    wrapper::{StrongholdWrapper, WrapperError},
+};
+
+thread_local! {
+    static LAST_ERROR: RefCell<Option<Box<dyn Error>>> = RefCell::new(None);
+}
+
+fn push_error(err: WrapperError) {
```

Comment:
nit: It's more `set_last_error` than a push, since you only store the last error